src/sentry/static/sentry/app/views/settings/account/accountEmails.jsx

```diff
@@ -31,16 +31,18 @@ const RemoveButton = styled(({hidden, ...props}) => (
 class EmailRow extends React.Component {
   static propTypes = {
     email: PropTypes.string.isRequired,
+    onRemove: PropTypes.func.isRequired,
+    onVerify: PropTypes.func.isRequired,
     isVerified: PropTypes.bool,
     isPrimary: PropTypes.bool,
     hideRemove: PropTypes.bool,
-    onRemove: PropTypes.func,
-    onVerify: PropTypes.func,
     onSetPrimary: PropTypes.func,
   };
 
   handleSetPrimary = e => {
-    this.props.onSetPrimary(this.props.email, e);
+    if (typeof this.props.onSetPrimary === 'function') {
```

Comment:
i don't really understand why being defencive is helpful here - don't we want to see an error thrown if this happens? or is there a known case where it setPrimary wouldn't be defined?

Reply:
I'll revert in another PR, don't want to go through another travis cycle